src/pages/ClusterPage/new.js

```diff
@@ -210,7 +210,7 @@ export default class NewCluster extends Component {
 		let pattern = /^[a-zA-Z0-9]+([-]+[a-zA-Z0-9]*)*[a-zA-Z0-9]+$/;
```

Comment:
The general pattern as well can avoid using the last `[a-zA-Z0-9]+` condition, I think so.